README.md

```diff
@@ -86,6 +86,18 @@ Currently, supported relations are:
 - `loki-config`, for interfacing with [loki][Loki operator].
 - `grafana-dashboards`, for interfacing with [grafana][Grafana operator].
 
+
+### About Juju Topology
+
+This charm forwards alert rules, recording rules and dashboards but does not add its own metadata to the topology.
+
+The [Juju topology](https://charmhub.io/observability-libs/libraries/juju_topology) describes a node in the model, not the data flow. That's why this charm does not inject Juju topology.
+
+Since a cos-configurator _app_ needs to be deployed per app, topology labels _could_ be used to give a since of origin (as in data flow). However, this would be semantically inconsistent with charms, where topology labels indicate a node in the model. This may unpredictably interfere with `group_by` directive.
```

Comment:
```suggestion
While a cos-configuration charm provides alerting rules, recording rules, and dashboards for charms, and topology labels _could_ be used to give a since of origin (as in data flow), the cos-configuration _deployment_ itself is neither enriched with nor aware of suitable values for metadata to identify workloads. 

In addition, the ability of `cos-configuration` to provide rules and dashboards which are not intrinsically tied to topology metadata offers administrators the flexibility to use COS to monitor non-charmed applications, use rules or dashboards directly from other sources, implement aggregate dashboards or rules which may collate metrics from more than one application, and more.

Addition of Juju topology metadata to the data structures provided by cos-configuration would be semantically inconsistent with charms, where topology labels indicate a node (application or unit) in Juju, and cos-configuration itself would not be consistent with the design model of Juju topology if it were to suggest label selectors for applications whose status cannot be known by cos-configuration itself. 

Finally, addition of Juju topology labels may unpredictably interfere with `group_by` directive if an incorrect selector were injected.
```

Reply:
Amazin, thanks Ryan!